temel_bilgiler/giriş/web-geliştirmeye-giriş.md

```diff
@@ -0,0 +1,97 @@
+
+Backend, uygulamanın iç kısmını ifade eder, bu kısım sunucuda bulunur. Backend, ön uçun ihtiyaç duyduğu şeyleri sağlamak için program verilerini depolar ve sunar. Bu süreç, bir website'nin milyonlarca kullanıcısı olduğunda çok karmaşık hale gelebilir. Backend geliştiricileri veri ile çalışmak için Java, Python, Ruby ve JavaScript gibi programlama dillerini kullanır.
+
+FullStack geliştiriciler hem ön uç hem de arka uç ile rahat çalışırlar. The Odin Project'te, web geliştirme ile ilgili tüm yönleri kapsayan tam yığın geliştirme öğretmeye odaklanıyoruz.
+
+Daha fazla detay için, Udacity'nin bu konu hakkında harika bir blog yazısı var:[3 Web Dev kariyer çözümlemesi: Front-End vs Back-End vs Full Stack](https://www.udacity.com/blog/2020/12/front-end-vs-back-end-vs-full-stack-web-developers.html)
+
+### Kariyer Türleri
+
+Şimdi farklı web geliştirme türlerini bildiğinize göre, daha önce bahsettiğimiz farklı müşteri ve işveren türlerini kapsayan şeylerden bahsedelim.
+
+Büyük teknoloji şirketleri, Google, Meta ve Amazon gibi, çok sıkı işe alım gereksinimlerine sahiptir. Bu beklentileri başarıyla karşılarsanız, mükemmel maaş, yan haklar ve fırsatlar sunarlar.
+
+Küçük teknoloji şirketleri, startup'lar, biraz vahşi batıya benzer. Bir junior geliştirici için, geliştirme hızı nedeniyle bir ateşleme denemesi gibi hissedebilir. Startup'lar genelde biraz daha düşük maaş sunar ve daha uzun saatler gerektirir, ama aynı zamanda şirkett hisseleri ve benzersiz çalışma ortamları sunabilirler.
+
+Freelanse olarak çalışmak, iyi bir saatlik ücret ve kendi ürünlerinizi planlama ve tasarlama özgürlüğüne sahip olmanızı sağlar. Ancak, kendi işinizi bulmak (bu da daha az kodlama zamanı anlamına gelebilir), müşterilerden faturalandırmayı yönetmek (bu süreç zorlayucı olabilir) için tek sorumlu siz olursunuz. Bu yol için güçlü insan becerilerine ihtiyacınız var.
+
+Bir web danışmanlık şirketi için çalışmak, freelance ücret potansiyelinizin bir kısmından vazgeçmenizi ama kod üzerinde daha fazla odaklanmanızı sağlar. Bu seçenek aynı zamanda iyi bir iş/yaşam dengesi ve ücret sunar.
+
+Son olarak, büyük, eski şirketler hala web geliştiricilere ihtiyaç duyar. Bu şirketler iyi bir iş/yaşam dengesi, ücret ve yan haklar sunar, ancak teknolojiye çok odaklanan bir şirketten daha yavaş aksiyon alabilirler.
+
+### Araçlar
+
+Aşağıda yaygın olarak kullanılan bazı araçlar var. Şu anda ne olduklarını bilmeyebilirsiniz, ancak ileride kesinlikle bileceksiniz.
+
+- **Bilgisayar**
+- **[Google](https://www.google.com/)**
+- **Text Editor**
+- **Command Line Interface (CLI)**
+- **[Stack Overflow](http://stackoverflow.com/)**
+- **[Git](https://git-scm.com/)**
+- **[GitHub](https://github.com/)**
+
+### Motivation
+
+Kod yazmayı öğrenmek inanılmaz derecede ödüllendirici olduğu kadar zor ve sinir bozucu da olabilir. Bir öğrenci olarak sahip olabileceğiniz en güçlü özellik, bir şeyler inşa etme isteği, problem çözme yeteneği ve engellerle karşılaştığınızda ısrarla üstüne gitmektir.
```

Comment:
fix: Kod yazmayı öğrenmek inanılmaz derecede ödüllendirici olduğu kadar zor ve sinir bozucu da olabilir. Bir öğrenci olarak sahip olabileceğiniz en güçlü özellikler; bir şeyler inşa etme isteği, problem çözme yeteneği ve engeller karşısında kararlı olmaktır.